nodejs/week3/index.js

```diff
@@ -0,0 +1,72 @@
+const { response } = require("express");
+const express = require("express");
+const app = express();
+app.use(express.urlencoded({ extended: true }));
+
+app.get("/calculator/:operator", (req, res,) => {
+    //make variable for path parameter to use multiple paths
+    var operator = req.params.operator;
+    const firstNumbers= req.query.first;
+    const secondNumbers= req.query.second;
+    //make array of values from both query parameters
+    const numsArr1 = Array.from(firstNumbers, Number);
+    const numsArr2 = Array.from(secondNumbers, Number);
+    //here we have two arrays now make these two into one array
+    const allNumbersArray = numsArr1.concat(numsArr2)
+    
+    let data;
+    //now make conditional statements to perform different function with different path parameters the help of operator variable
+    if(operator === "add"){
+        data = allNumbersArray.reduce((a, b)=> a+b, 0)
+
+    }
+    if(operator === "multiply"){
+        data = allNumbersArray.reduce((a, b)=> a*b, 1)
+
+    }
+    if(operator === "divide"){
+        data = allNumbersArray.reduce((a, b)=> a/b, 1)
+    }
+    if(operator === "subtract"){
+        data = allNumbersArray.reduce((a, b)=> a - b)
+    }
+    
+     res.send(`Total is: ${data}`);
+    
+    
+});
+app.post("/calculator/:operator", async (req, res) => {
+    var operator = req.params.operator;
+    const firstNumber = Number(req.body.first)
+    const secondNumber = Number(req.body.second)
+    
+    let data;
+    if(operator === "add"){
+        data =firstNumber + secondNumber
```

Comment:
One idea to improve your code and make it more reusable is to create 4 functions (not routes) eg (addNumbers, subtractNumbers..) which do the operations required and use those functions both in the GET and the POST request.
That way, you can use the same code and avoid re-writing logic for adding/subtracting numbers.

Reply:
hi thank you for your feedback. i am performing reduce function on allNumbersArray(line15) to add subtract and multiply. because in .get router we are passing infinite number of parameters. but in post we only using two parameters. hence there is no Array as such to perform the same reduce function.

Reply:
if in post router we want to pass same parameters again and again with different values. then  I have to get all the number into two different arrays then make them into one array like I did in .Get router. so now please suggest me what should I do????

Reply:
Well, think about this: Does the GET route work when you pass only two parameters and not an infinite number? If so, why couldn't the POST route work in the same way?

What if you create an allNumbersArray from the two parameters of the POST route and pass it to an ```addNumbers(numbersArray)``` function which uses reduce?

Reply:
in that case I have to repeat everything same to same from line 8 to line 15. then it is also repeating the same code again. I tried to put this into a function and try to reuse it but its not working. another thing all 4 reduce functions are doing different calculations like add, multiply, etc then how can I make them into one. I don't know what exactly to do...

Reply:
you can probably explain it to me on how could I minimise all my code and not repeat myself. because when I try not to repeat one part of the code the other part of the code gets repeated. so please show me how to achieve it.

Reply:
Yes you're right you would have to repeat that part of the code.
The most important part though that's better to not be repeated is the four operations. So,  you can just create 4 functions one for each operation and reuse those. It's fine if you repeat the code that creates the array or reads parameters etc.
The reason it's more important, is that it's where the main logic exists. So, say for instance in the future it is required to also round the numbers before adding them or doing something else before. Instead of writing the new code that rounds them in two places you'll only do it inside the addNumbers function.
This is the main point of this, reusing code so when the time comes you need to change something in your logic you do it in the fewest places possible

Reply:
hi thank you for your feedback. I have now made four functions based on math. operations and used them in my GET and Post requests. please have a look and let me know If I can merge the pull request. thank you once again.

Reply:
Looks great!! 
One very minor thing, in the four functions better to use 'array' instead of 'Array' for the name, since it is just a parameter to the function.
Great work it's ready 🚀